File: apps/web/src/types/faculty.ts
export interface Faculty {
  id: string;
  full_name: string;
  email: string;
  phone: string | null;
  department_id: string | null;
  designation: string;
  employment_type: "full_time" | "part_time" | "visiting" | "contract";
  joining_date: string;
  status: "active" | "on_leave" | "resigned" | "retired";
}

export interface FacultyListResponse {
  data: Faculty[];
  meta: { page: number; page_size: number; total: number };
}

export interface FacultyCreateInput {
  full_name: string;
  email: string;
  phone?: string | null;
  department_id?: string | null;
  designation: string;
  employment_type?: Faculty["employment_type"];
  joining_date: string;
}

export type FacultyUpdateInput = Partial<
  Pick<FacultyCreateInput, "full_name" | "phone" | "designation"> & { status: Faculty["status"] }
>;

File: apps/api/app/schemas/faculty.py
import uuid
from datetime import date

from pydantic import BaseModel, ConfigDict, Field


class FacultyBase(BaseModel):
    full_name: str = Field(min_length=1, max_length=255)
    email: str
    phone: str | None = None
    department_id: uuid.UUID | None = None
    designation: str = Field(min_length=1, max_length=128)
    employment_type: str = Field(
        default="full_time", pattern="^(full_time|part_time|visiting|contract|intern)$"
    )
    joining_date: date


class FacultyCreate(FacultyBase):
    pass


class FacultyUpdate(BaseModel):
    full_name: str | None = None
    phone: str | None = None
    designation: str | None = None
    status: str | None = Field(
        default=None, pattern="^(active|on_leave|resigned|retired)$"
    )


class FacultyRead(FacultyBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    status: str


class PaginationMeta(BaseModel):
    page: int
    page_size: int
    total: int


class FacultyListResponse(BaseModel):
    data: list[FacultyRead]
    meta: PaginationMeta
